core/win32/ntdll_shared.c

```diff
@@ -54,7 +54,9 @@
 #    pragma warning(disable : 4204) // nonstd extension: non-constant aggregate
                                     // initializer
 #    define INVALID_FILE INVALID_HANDLE_VALUE
-#    define snprintf _snprintf
+#    if defined(_MSC_VER) && _MSC_VER < 1900
+#        define snprintf _snprintf
```

Comment:
The commit title makes it sound like doing nothing does not cause any problem and this is just cleanup.  Is that the case?  Or is there some compiler warning or error without this change?

Reply:
> Or is there some compiler warning or error without this change?

Yes, as I'm working on DR patch for MSVS 2015+ support, I've encountered such an issue.

`stdio.h:1927` from Win SDK 10.0.17763.0 contains the following code:
```c++
#if defined snprintf
        // This definition of snprintf will generate "warning C4005: 'snprintf': macro
        // redefinition" with a subsequent line indicating where the previous definition
        // of snprintf was.  This makes it easier to find where snprintf was defined.
        #pragma warning(push, 1)
        #pragma warning(1: 4005)
        #define snprintf Do not define snprintf as a macro
        #pragma warning(pop)
        #error Macro definition of snprintf conflicts with Standard Library function declaration
#endif
```
Definitely it gives a warning about redefinition of `snprintf` then aborts compilation.